src/deploy/dataconnect/deploy.ts

```diff
@@ -88,22 +88,23 @@ export default async function (
         return !filters || filters?.some((f) => si.dataConnectYaml.serviceId === f.serviceId);
       })
       .map(async (s) => {
-        const instanceId = s.schema.primaryDatasource.postgresql?.cloudSql.instance
-          .split("/")
-          .pop();
-        const databaseId = s.schema.primaryDatasource.postgresql?.database;
-        if (!instanceId || !databaseId) {
-          return Promise.resolve();
+        const primaryDatasource = s.schema.datasources.find((d) => d.postgresql);
```

Comment:
nit: For future-proofing (since the "primary" datasource won't necessarily be the Postgres one) would it make more sense to name this `postgresDatasource`?